endpoints/openrtb2/amp_auction.go

```diff
@@ -698,3 +731,20 @@ func setEffectiveAmpPubID(req *openrtb2.BidRequest, account string) {
 		pub.ID = account
 	}
 }
+
+func setTrace(req *openrtb2.BidRequest, value string) (err error) {
+	if value == "" {
+		return nil
+	}
+
+	ext := json.RawMessage(`{"prebid":{"trace":"` + value + `"}}`)
```

Comment:
Please switch to using `json.Marshal` here to ensure there the `value` string is not an attack vector.